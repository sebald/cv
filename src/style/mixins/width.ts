import { InterpolationFunction } from '../theme';
import { px } from '../utils';

export const width: InterpolationFunction<{ width?: number }> = ({
  width: w,
}) => ({
  ...(w ? { width: w <= 1 ? `${w * 100}%` : px(w) } : {}),
});
