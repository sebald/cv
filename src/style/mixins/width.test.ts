import { width } from './width';

test('with mixin', () => {
  expect(width({ width: 0.5 })).toMatchSnapshot();
  expect(width({ width: 0.25 })).toMatchSnapshot();
  expect(width({ width: 1 })).toMatchSnapshot();

  expect(width({ width: 100 })).toMatchSnapshot();
  expect(width({ width: 800 })).toMatchSnapshot();
});
