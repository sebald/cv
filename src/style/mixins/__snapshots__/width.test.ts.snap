// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with mixin 1`] = `
Object {
  "width": "50%",
}
`;

exports[`with mixin 2`] = `
Object {
  "width": "25%",
}
`;

exports[`with mixin 3`] = `
Object {
  "width": "100%",
}
`;

exports[`with mixin 4`] = `
Object {
  "width": "100px",
}
`;

exports[`with mixin 5`] = `
Object {
  "width": "800px",
}
`;
